client.go

```diff
@@ -142,6 +143,70 @@ func newClient(zkquorum string, options ...Option) *client {
 	return c
 }
 
+// DebugState information about the clients keyRegionCache, and clientRegionCache
+func DebugState(client Client) ([]byte, error) {
+
+	debugInfoJson, err := json.Marshal(client)
+	if err != nil {
+		log.Errorf("Cannot turn client into JSON bytes array: %v", err)
+	}
+	return debugInfoJson, err
+}
+
+func (c *client) DoneStatus() string {
```

Comment:
No need to export (capitalize) this function. Since it's only used by MarshalJSON this code could just be moved in that function instead